### ISSUE TITLE
Correction: 'NA' are valid values for some columns. Categorical dummies are not getting formed for them.

### ISSUE THREAD

**Issue by Gaurav-Sachdeva** [tarun-goyal/House-Prices#12]

**Comment by tarun-goyal** [comment]
@Gaurav-Sachdeva 
I checked this bug at my end. Did not get any such issue in the cleaned design matrices (both training and test matrix). Each of the categorical features was taking `None` as a legitimate value and forming a dummy for them - `Cat_feature_None`. Attached are the transformed files after data cleaning exercise:

[test_design_matrix.xlsx](https://github.com/tarun-goyal/House-Prices/files/788999/test_design_matrix.xlsx)
[train_design_matrix.xlsx](https://github.com/tarun-goyal/House-Prices/files/789000/train_design_matrix.xlsx)

Could you again check at your end if you still face this issue.

**Comment by tarun-goyal** [comment]
Closing this as I believe this is no longer an issue.

**Comment by Gaurav-Sachdeva** [comment]
oh sorry Tarun I forgot to check it earlier. In my system, the design matrix still does not contain dummy for NA value. Anyways, apke system pe thik h to we can ignore. Mostly good solutions apne hi submit kia hain.

1 more thing. In the above **test_design_matrix.xlsx** that you shared, there are categorical dummies getting formed for a continuous variable named "BsmtUnfSF" (which we were not imputing). I checked into it and saw that there is a NA value in this column which represents "No Basement" (House ID: 2121). So though its a continuous variable, one of its row is categorical. 

Since my system is treating valid NA as empty, when I run clean_data function on test data, no categorical dummies are getting formed for this variable, i.e. its still continuous but with a one missing value (which actually is not a missing, but a valid NA value). 

However, since your system is treating NA not as a missing, but as a valid categorical value, I think if you run clean_data function on test data, it would still result in dummy columns for that variable. If you can check that.

Currently I'm not sure how to treat a mixed continuous and categorical column. Should we convert this NA to zero.

Same thing happening for a few other basement related columns: BsmtFinSF2, TotalBsmtSF, etc

**Comment by tarun-goyal** [comment]
Good catch :)

Here are the definitions of these variables:

BsmtFinSF2: Type 2 finished square feet
BsmtUnfSF: Unfinished square feet of basement area
TotalBsmtSF: Total square feet of basement area

If `BsmtFinType1` or `BsmtFinType2` represents 'No Basement' for these missing points, then we should replace NA values with 0 as there is no basement at all.

**Comment by tarun-goyal** [comment]
You were right all the way regarding NA values. I ran only `data_cleansing.py` file and looked for the missing values in training and test data. However, I overlooked the scenario of this happening while `XGB_GridSearch.py` or `XGBoosting.py` were executing.

The code considered 'None' as the valid value when I ran `data_cleansing.py` file, but when this code was called from outside the module, 'None' values were no longer valid and considered as missing.

Working on the correction now. Thanks for replying to this issue, and again bringing my attention to the data set :)
